execution_engine/src/execution/error.rs

```diff
@@ -177,6 +179,12 @@ pub enum Error {
     /// Transform error.
     #[error(transparent)]
     Transform(transform::Error),
+    /// Invalid key
+    #[error("Invalid key {0}")]
+    InvalidKey(Key),
+    /// Invalid Contract package kind.
+    #[error("Invalid contract package kind")]
```

Comment:
```suggestion
    #[error("Invalid contract package kind: {0}")]
```
for better error reporting (needs `impl Display`)